src/autopas/tuning/AutoTuner.cpp

```diff
@@ -273,22 +275,31 @@ void AutoTuner::addMeasurement(long sample, bool neighborListRebuilt) {
   }
 }
 
-void AutoTuner::bumpIterationCounters() {
-  ++_iteration;
+void AutoTuner::bumpIterationCounters(bool needToWait) {
+  // reset counter after all autotuners finished tuning
+  if (not(needToWait or inTuningPhase() or _iterationsSinceTuning < _tuningInterval)) {
+    _iterationsSinceTuning = 0;
+  }
   ++_iterationsSinceTuning;
-  // this will NOT catch the first tuning phase because _iterationsSinceTuning is initialized to _tuningInterval.
-  // Hence, _tuningPhase is initialized as 1.
```

Comment:
It confused me more than it helped tbh.  because `_tuningPhase` is initialized as 0. If you insist, I can get back the first line of the  comment, but I guess this will change anyways with #943